### ISSUE TITLE
General creatures issue

### ISSUE THREAD

**Issue by hwoarangmy** [OpenDungeons/OpenDungeons#1073]
This issue is for listing the creatures per faction and discuss which ones should be replaced when we have new models. I will list the ones we currently use per tier. I will try to update this thread when new creatures are integrated.

The order from left to right corresponds to "better than". Because of the triangle, the creature on the right is better than the creature on the left.
I will also add a "(T)" for temporary creatures (ie not final - to be replaced by better models) based on @Danimal696  list:
http://forum.freegamedev.net/viewtopic.php?f=35&t=5961&sid=f08966b36cc2dbb5d8f6c4d238918a9b#p66511

Keeper side:
Tier1: Spider, CaveHornet, Rat
Tier2: LizardMan, Roach, Goblin
Tier3: DarkElf, Kreatur (T), Orc (T)
Tier4: Dragon (T), Troll, PitDemon (T)
Worker: Kobold (T)

Hero side:
Tier1: CaveHornet, Slime, Dwarf1 (T)
Tier2: Adventurer, Knight (T), Dwarf2 (T)
Tier3: Elf, RunelordDwarf, Wizard (T)
Tier4: NatureMonster, Defender, Scarab
Worker: Dwarf (T)

Special (spawn from room): Skeleton, Lich (T)

Rogue creatures are creatures that can be added to any level like the other ones but that do not spawn through portals (they are defined in creatures.cfg file but not in factions.cfg and spawnconditions.cfg)
Rogues: Wyvern (T), gnome (T), TentacleAlbine, TentacleGreen

**Comment by hwoarangmy** [comment]
IMHO, some of them could be considered as final: Goblin, LizardMan, Orc, Kreatur, Knight, Wizard.
Actually, I'm surprized Goblin is not considered final. @Danimal696 could you please confirm the list of final creatures in your mind ?
Considering dwarves, I would say they are ok and not the most in need of love.

I have integrated a new creature "Rat" in my last PR and I propose to replace the wyvern.

@OpenDungeons/developers WDYT ? Are you ok with this list ?

**Comment by Bertram25** [comment]
> This issue is for listing the creatures per faction and discuss which ones should be replaced when we have new models. I will list the ones we currently use per tier. I will try to update this thread when new creatures are integrated.

Perfect, thanks.

> @OpenDungeons/developers WDYT ? Are you ok with this list ?

Could it be possible to also update the wiki page there: https://github.com/OpenDungeons/OpenDungeons/wiki/Creatures
With the corresponding image next to the creature status, it's easier to discuss, at least to me.

Btw, I suppose you forgot about workers, kobold, dwarf, the robot (is it still there?), the Elf, the Lich and skeleton. (We should list them all even if they are not part of the official spawnable list, right?)

To me, both base workers deserve replacement.
I would replace the slime if possible (as a low priority) as the creature is kinda boring.
I would also replace the tentacle for same boring guilt. ;)

Apart from that, I'm fine with replacing the ones proposed, but I would insist that evil creatures should be impressive or fun, even at Tier I (well, less impressive than others) so that the player is "happy" with his/her army. And the hero part could be either heroic or not reassuring. It will give some more immersion to the player IMHO.

Thanks again for this effort in the right direction. :)

**Comment by hwoarangmy** [comment]
> Btw, I suppose you forgot about workers, kobold, dwarf, the robot (is it still there?), the Elf, the Lich and skeleton. (We should list them all even if they are not part of the official spawnable list, right?)

Yes, I was talking about spawnable units only. But you are right, now that we are there, we should list them all

> To me, both base workers deserve replacement.

Last excuse for @Danimal696 to not replace the kobold was to be in the montain. Ridiculous, isn't it ?

> I would replace the slime if possible (as a low priority) as the creature is kinda boring.
> I would also replace the tentacle for same boring guilt. ;)

I'm okay with it. Same kind of impression as the dwarf for me.

> Apart from that, I'm fine with replacing the ones proposed, but I would insist that evil creatures should be impressive or fun, even at Tier I

Well, I guess we all agree but finding good models is not that easy ;-)
Let's not whip @Danimal696 for this (unless you really want as he likes it anyway)

**Comment by Bertram25** [comment]
> Last excuse for @Danimal696 to not replace the kobold was to be in the montain. Ridiculous, isn't it ?

Arf, just let me kindle the fire in the torture room again...

> Well, I guess we all agree but finding good models is not that easy ;-)
> Let's not whip @Danimal696 for this (unless you really want as he likes it anyway)

Owww, ok. (/me stops the fire just started in the torture room.)
More serioulsly, sure, no pressure as I know it's a lot of work. Just that if we do replacement, let's aim for something really inline with that as much as possible. :)

**Comment by Danimal696** [comment]
> Last excuse for @Danimal696 to not replace the kobold was to be in the montain. Ridiculous, isn't it ?

Im starting my own dungeon, but thats a secret, okay? ;) i got some lost climbers chained and digging in there for the time being...

Goblin is already final, for lizardman i only plan a skin and animation redo, orc needs new model as mostly all heroes creatures...

> as the creature is kinda boring.

Lets keep them as mobs even if they are not used as creatures in the future, finding nice monster is pretty hard, and its even worse if they have to be more unique.

**Comment by Bertram25** [comment]
> Lets keep them as mobs even if they are not used as creatures in the future, finding nice monster is pretty hard, and its even worse if they have to be more unique.

Good point. It's true we can keep spare creatures model as rogue ones for more variety. The tentacle as a standard mob is fine to me. :)

@hwoarangmy Let's also add the rogue creatures in our creature catalog if it's ok for you. :)

**Comment by akien-mga** [comment]
I also thought about growing a pool of "unique" rogue creatures (i.e. creatures which do not belong to the other factions) so that we can add more diversity to the levels. So we should not be too eager to drop a creature as soon as we introduce a new one IMO, unless it's clear to everyone that the old creature really isn't cool enough :)

But that's exactly the purpose of this topic, so thanks for starting it hehe.

**Comment by hwoarangmy** [comment]
I've added the creatures we talked about. I've removed the temporary for lizardman as temporary creatures are creatures to be replaced.
I've also tagged the lich as temporary because it is not the best model we have but it is acceptable IMHO.

I've also moved the wyvern to rogue as it is replaced by the rat in the spawnable list

**Comment by Danimal696** [comment]
who is gonna research now?

**Comment by hwoarangmy** [comment]
well, ATM, it is the rat :-(
It has the same stats as the wyvern. Later, I guess we could use the lizardman as a level 2 creature if we find something bigger (and more evil) for a tier 3

**Comment by hwoarangmy** [comment]
@Danimal696 I've just thought that I didn't use the Elf/DarkElf yet while they are integrated. I could also replace lizardman/knight as tier3 creatures by the Elves.
Lizardman would become tier 2 reasearcher instead of the rat.
And rat could replace tier 1 gnome for the hero faction/
WDYT ?

**Comment by akien-mga** [comment]
I wouldn't put the rat in the hero faction, if he doesn't fit for the keeper IMO it would be a nice rogue creature (nobody likes rats after all).

**Comment by hwoarangmy** [comment]
IMHO, we should try to have "final" creatures for factions as soon as possible. If all our models were good enough, I wouldn't say the same but using a model of such quality for rogue creatures seems a waste for me and it is one less model for @Danimal696 to do.

what about moving slime to hero side (instead of gnome) and rat as tier 1 for keeper ?

**Comment by Danimal696** [comment]
slimes are not that heroic, but since all is just temporary i think its fine; when someday we fill more creatures we will decide on a final setup

**Comment by hwoarangmy** [comment]
ok. Then, I propose:
Keeper side:
Tier1: Spider, CaveHornet, Rat
Tier2: LizardMan, Roach, Goblin
Tier3: DarkElf, Kreatur (T), Orc (T)
Tier4: Dragon (T), Troll, PitDemon (T)

Hero side:
Tier1: CaveHornet, Slime, Dwarf1 (T)
Tier2: Adventurer, Knight (T), Dwarf2 (T)
Tier3: Elf, RunelordDwarf, Wizard (T)
Tier4: NatureMonster, Defender, Scarab

Tentacle and gnome becoming rogue

I will update my PR if nobody disagrees

**Comment by Bertram25** [comment]
> Keeper side:
> Tier1: Spider, CaveHornet, Rat
> Tier2: LizardMan, Roach, Goblin
> Tier3: DarkElf, Kreatur (T), Orc (T)
> Tier4: Dragon (T), Troll, PitDemon (T)

Fine as for me.

> Hero side:
> Tier1: CaveHornet, Slime, Dwarf1 (T)
> Tier2: Adventurer, Knight (T), Dwarf2 (T)
> Tier3: Elf, RunelordDwarf, Wizard (T)
> Tier4: NatureMonster, Defender, Scarab

I would set Slime and CaveHornet as temprorary for Heroes for now, I would replace them for more typical heroic creatures asap. Overall, I'd propose this:

Hero side:
Tier1: Adventurer, Slime (T), Dwarf1 (T)
Tier2: _HeroicAnimal2_ TBD, Knight (T), Dwarf2 (T)
Tier3: Elf, RunelordDwarf, Wizard (T)
Tier4: NatureMonster, Defender, Scarab

**Comment by hwoarangmy** [comment]
@Bertram25 Note that the (T) tag do not mean temporary in team but temporary model (the model is to be replaced).
Concerning hero faction, it has been built with the available creatures and I guess we all agree that many do not look "heroic" enough. However, until we have better models, let's keep them for having more variety ^^

**Comment by akien-mga** [comment]
Should we maybe distinguish between a long-term plan (what we'd like the factions to be like for 1.0, based on both current models and models that we would like to import) and a short-term plan (what we want to use for 0.7 and maybe 0.8)? The short-term plan would evolve little by little to try to reach the long-term one :)

**Comment by hwoarangmy** [comment]
I'm not sure we can really go for long term plans because AFAIK, our model choice is driven by the models we can find. If we find something nice for keeper, we try to remove another one that could fit in hero side and vice versa

**Comment by akien-mga** [comment]
> If we find something nice for keeper, we try to remove another one that could fit in hero side and vice versa

But that's precisely what I think we should try not to do, if we swap creatures from one faction to the other each time we introduce a new model, we'll have a hard time defining factions which are really distinct from each other. IMO we need to really accentuate the evil part of the keeper and the "heroic fairy tale nice guys" part of the heroes. That's why I think we need a long term plan to have an idea of the direction we're going to. If we see we lack many creatures for the long term plan, then we know what to look for. And if we see that some of the creatures we currently use for one faction don't fit into the long term plan, then we'll know they should either fit as rogue creatures, special campaign creatures or will end up removed when we get better models.

**Comment by hwoarangmy** [comment]
Then, you are looking for a very looonng term plan ;-)
Unless you whip @Danimal696 enough ^^

**Comment by akien-mga** [comment]
> Then, you are looking for a very looonng term plan ;-)

Well yes, for the long term plan, I mean long. As in, we take a look at each creature that we currently have, and we try to think realistically: where does it fit? Which factions (keeper, hero, rogue - the latter would likely also include "special" creatures that are not spawnable but used in some campaigns for example), what kind of characters, and what tier.

This long term plan would have its own issue separate from the short term "what do we do with our current creatures for the next release?" issue, so that we can think a bit more freely about where we're going. I think we need this kind of long term plans if we want to go towards a game that feels polished and well thought, and not just a patchwork of what we found on blendswap and had the time to adapt to our game :)

I don't mean that how we're doing currently is bad, the long term approach is complementary IMO.

**Comment by hwoarangmy** [comment]
> This long term plan would have its own issue separate

Yes. I was going to say the same ^^

**Comment by hwoarangmy** [comment]
Now, we have 3 new creatures:
- Cultist
- Lava Spawn
- Monk

I propose:
Keeper side:
Tier1: Spider, CaveHornet, Rat
Tier2: LizardMan, Roach, Goblin
Tier3: DarkElf, Cultist, Orc (T)
Tier4: LavaSpawn, Troll, PitDemon (T)

Hero side:
Tier1: Adventurer, Slime, Dwarf1 (T)
Tier2: Monk, Knight (T), Dwarf2 (T)
Tier3: Elf, RunelordDwarf, Wizard (T)
Tier4: NatureMonster, Defender, Scarab

Dragon and Kreatur would be rogue

@OpenDungeons/developers  WDYT ?

**Comment by Loobinex** [comment]
I really like the lava spawn monster, but I'd rather not see it as a T4 creature, but T2 or T3 would be fine. However I think it would fit best as a rogue to be places around lava pits. It could also get a blue pallet swap to become a frost-spawn for ice maps.

I also have my doubts about the Cultist on T3, because thematically it looks like the main researcher on the evil side and research won't wait for T3.

**Comment by hwoarangmy** [comment]
> I really like the lava spawn monster, but I'd rather not see it as a T4 creature, but T2 or T3 would be fine.

Yes, I agree it doesn't look like a T4 unit. It doesn't look strong enough. But I think we should try to replace temporary units and Dragon is one of our weakest models.

> I also have my doubts about the Cultist on T3, because thematically it looks like the main researcher on the evil side and research won't wait for T3.

You have a point here.

Maybe we can go for:
Keeper side:
Tier1: Spider, CaveHornet, Rat
Tier2: Cultist, Roach, Goblin
Tier3: DarkElf, LizardMan, LavaSpawn
Tier4: Orc (T), Troll, PitDemon (T)

Hero side:
Tier1: Adventurer, Slime, Dwarf1 (T)
Tier2: Monk, Knight (T), Dwarf2 (T)
Tier3: Elf, RunelordDwarf, Wizard (T)
Tier4: NatureMonster, Defender, Scarab

**Comment by Loobinex** [comment]
Better.

And keep open the possibility when you make/find another good monster to turn the LavaSpawn rogue and replace it with another T3 creature.

**Comment by hwoarangmy** [comment]
Well, if we find better models, I guess any one can be removed to be rogue only ^^
